src/app/upgrade.cpp

```diff
@@ -75,10 +85,66 @@ namespace
             , QLatin1String("Preferences/WebUI/HTTPS/KeyPath")
             , Utils::Fs::toNativePath(configPath + QLatin1String("WebUIPrivateKey.pem")));
     }
+
+#if defined Q_OS_WIN
+    void upgradeResumeFiles()
+    {
+        QString backupFolderPath = Utils::Fs::expandPathAbs(specialFolderLocation(SpecialFolder::Data) + "BT_backup");
+        QDir backupFolderDir(backupFolderPath);        
+
+        const QStringList backupFiles = backupFolderDir.entryList(QStringList(QLatin1String("*.fastresume")), QDir::Files, QDir::Unsorted);
+        const QRegularExpression rx(QLatin1String("^([A-Fa-f0-9]{40})\\.fastresume$"));
+        for (const QString &backupFile : backupFiles) {
+            const QRegularExpressionMatch rxMatch = rx.match(backupFile);
+            if (rxMatch.hasMatch()) {
+                const QString hashStr = rxMatch.captured(1);
+                const QString filepath = backupFolderDir.absoluteFilePath(backupFile);
+
+                QFile file1(filepath);
+                if (!file1.open(QIODevice::ReadOnly))
+                    continue;
+
+                QByteArray data = file1.readAll();
+                file1.close();
+
+                libtorrent::error_code ec;
+                libtorrent::bdecode_node fastOld;
+                libtorrent::bdecode(data.constData(), data.constData() + data.size(), fastOld, ec);
+                if (ec || (fastOld.type() != libtorrent::bdecode_node::dict_t))
+                    continue;
+
+                libtorrent::entry fastNew;
```

Comment:
Seems you don't really need it.

Reply:
Tries to use `fastOld`, but seems like it doesn't work the same way as `libtorrent::entry`.